shared/test/test_files.rb

```diff
@@ -344,6 +344,38 @@ def test_bogus_metadata
     assert not_found?
   end
 
+  def test_rename_operations
```

Comment:
As I understand it, this is testing a rename with overwrite? Perhaps name the test as such, introducing a vanilla `test_rename_operation` test as well.